users/urls.py

```diff
@@ -15,6 +21,7 @@
     url(r'^logout/$', views.UserLogoutView, name='logout'),
     url(r'^update-password/$', views.UserPasswordView, name='update-password'),
     url(r'^(?P<pk>\d+)/$', UserProfileView.as_view(), name='profile'),
+    url(r'^(?P<username>\w+)/$', views.UserProfileViewByUsername, name='profile_username'),
```

Comment:
This URL scheme is problematic if there is a user named `2` and a user with ID `2`.
In this case, the user with ID `2` is shown, as the `UserProfileView` matches before the `UserProfileViewByUsername` is checked.